x-pack/plugins/observability_solution/observability/public/components/alert_overview/helpers/get_sources.ts

```diff
@@ -36,7 +36,7 @@ export const getSources = (alert: TopAlert) => {
         const fieldValue = alertFields[field];
         matchedSources.push({
           field: source,
-          value: fieldValue[0],
+          value: Array.isArray(fieldValue) ? fieldValue[0] : fieldValue,
```

Comment:
yes, on alert details page, the links where generated with only first letter of the asset like `host.name: h` instead of `host.name: host-0`.

Reply:
Nice catch! 🐞 
I still don't understand why fieldValue is sometimes an array and sometimes a string, does it make sense to have a test for it?

Reply:
It looks like alert flyout doesn't parse the alert or use different parser, not sure. In alert flyout, all alert fields are arrays except _id and _index. I don't know which type of test would be useful in this scenario. I am inclined towards not adding test for this. If you have a suggestion for the type of test, do let me know.


<img width="573" alt="Screenshot 2024-04-12 at 18 40 02" src="https://github.com/elastic/kibana/assets/69037875/d629fb5c-02e3-40a7-91f2-9af028e48822">

Reply:
I was more thinking about a unit test but maybe we can improve types instead since I see we have some `any` and casting (`as AlertFields`) in this function.

```
  const matchedSources: Array<{ field: string; value: any }> = [];
  const ALL_SOURCES = [...infraSources, ...apmSources];
  const alertFields = alert.fields as AlertFields;
  ALL_SOURCES.forEach((source: string) => {
    Object.keys(alertFields).forEach((field: any) => {
```
Not for fixing in this PR, in general, having such loose types is dangerous.